web/src/settings_org.js

```diff
@@ -471,12 +476,17 @@ export function discard_property_element_changes(elem, for_realm_default_setting
         $elem,
         for_realm_default_settings,
     );
-    const property_value = settings_components.get_property_value(
-        property_name,
-        for_realm_default_settings,
-        sub,
-        group,
-    );
+    let property_value;
+    if (for_realm_default_settings) {
+        property_value =
+            settings_components.get_realm_default_setting_property_value(property_name);
+    } else if (group) {
+        property_value = settings_components.get_group_property_value(property_name, group);
+    } else if (sub) {
+        property_value = settings_components.get_stream_settings_property_value(property_name, sub);
+    } else {
+        property_value = settings_components.get_realm_settings_property_value(property_name);
+    }
```

Comment:
We should also split this function, like the others it's mainly a bunch of switch statements for individual settings and thus will do nicely being split up.